runtime/src/integration/bitcoin_simulator.rs

```diff
@@ -192,13 +192,28 @@ impl MockBitcoinCore {
     fn generate_coinbase_transaction(address: &BtcAddress, reward: u64, height: u32) -> Transaction {
         let address = ScriptBuf::from(address.to_script_pub_key().as_bytes().to_vec());
 
+        // construct height: see https://github.com/bitcoin/bips/blob/master/bip-0034.mediawiki
+        // first byte is number of bytes in the number (will be 0x03 on main net for the next
+        // 150 or so years with 223-1 blocks), following bytes are little-endian representation
+        // of the number (including a sign bit)
+        let mut height_bytes = height.to_le_bytes().to_vec();
+        for i in (1..4).rev() {
+            // remove trailing zeroes, but always keep first byte even if it's zero
+            if height_bytes[i] == 0 {
+                height_bytes.remove(i);
+            } else {
+                break;
+            }
+        }
+        height_bytes.insert(0, height_bytes.len() as u8);
```

Comment:
Can we not use the logic in the parsing crate here?

Reply:
I guess technically yes, but (1) it's not exposed and (2) we're not using the parachain's bitcoin types here, but rather the types used by the bitcoin core rpc. Both of these are not hard blockers but I think it makes it not worth it